### PR TITLE
fix(telemetry): ask only once to install telemetry when creating an sdk

### LINKED ISSUE #1408: [Bug]: npm create @ama-sdk@10.0.0-rc.7 with --spec-path asks twice whether to enable telemetry
### Package name

create

### Package version

10.0.0-rc.7

### Reproduction steps

Make sure you have an open api spec in `petstore.yaml` (for example, download it from [here](https://petstore3.swagger.io/api/v3/openapi.yaml)) and then run:
```
npm create --yes @ama-sdk@10.0.0-rc.7 typescript sdk/sdk -- --spec-path=petstore.yaml
```

### Current result

I get this question:

```
?
Would you like to share anonymous data about the usage of Otter builders and schematics with the Otter Team at Amadeus ?
It will help us to improve our tools.
For more details and instructions on how to change these settings, see https://github.com/AmadeusITGroup/otter/blob/main/docs/telemetry/README.md.
         (Y/n)
```

I answer no (n) and press enter.
Then the console shows this:

```
    You can activate it at any time by running `ng add @o3r/telemetry`.
    "6.6.0"
CREATE sdk/sdk/.commitlintrc.json (810 bytes)
CREATE sdk/sdk/.editorconfig (208 bytes)
CREATE sdk/sdk/.openapi-generator-ignore (90 bytes)
CREATE sdk/sdk/.renovaterc.json (1862 bytes)
CREATE sdk/sdk/.swcrc (329 bytes)
CREATE sdk/sdk/.versionrc.json (110 bytes)
CREATE sdk/sdk/CONTRIBUTING.md (118 bytes)
CREATE sdk/sdk/jest.config.js (725 bytes)
CREATE sdk/sdk/openapitools.json (241 bytes)
CREATE sdk/sdk/readme.md (7070 bytes)
CREATE sdk/sdk/tsconfig.build.json (542 bytes)
CREATE sdk/sdk/tsconfig.doc.json (329 bytes)
CREATE sdk/sdk/tsconfig.json (226 bytes)
CREATE sdk/sdk/typedoc.json (289 bytes)
CREATE sdk/sdk/.vscode/settings.json (613 bytes)
CREATE sdk/sdk/configs/tsconfig.test.json (496 bytes)
CREATE sdk/sdk/src/index.ts (77 bytes)
CREATE sdk/sdk/src/api/fixtures.jest.ts (123 bytes)
CREATE sdk/sdk/src/api/index.ts (17 bytes)
CREATE sdk/sdk/src/fixtures/jest/index.ts (41 bytes)
CREATE sdk/sdk/src/fixtures/jest/package.json (349 bytes)
CREATE sdk/sdk/src/helpers/index.ts (17 bytes)
CREATE sdk/sdk/src/helpers/package.json (310 bytes)
CREATE sdk/sdk/src/models/enums.ts (93 bytes)
CREATE sdk/sdk/src/models/index.ts (128 bytes)
CREATE sdk/sdk/src/models/patterns.ts (102 bytes)
CREATE sdk/sdk/src/models/base/enums.ts (97 bytes)
CREATE sdk/sdk/src/models/base/index.ts (17 bytes)
CREATE sdk/sdk/src/models/base/patterns.ts (100 bytes)
CREATE sdk/sdk/src/models/core/enums.ts (48 bytes)
CREATE sdk/sdk/src/models/core/index.ts (49 bytes)
CREATE sdk/sdk/src/models/core/patterns.ts (51 bytes)
CREATE sdk/sdk/src/models/custom/enums.ts (50 bytes)
CREATE sdk/sdk/src/models/custom/index.ts (17 bytes)
CREATE sdk/sdk/src/models/custom/patterns.ts (53 bytes)
CREATE sdk/sdk/src/spec/api-mock.ts (123 bytes)
CREATE sdk/sdk/src/spec/index.ts (103 bytes)
CREATE sdk/sdk/src/spec/operation-adapter.ts (16 bytes)
CREATE sdk/sdk/src/spec/package.json (289 bytes)
CREATE sdk/sdk/src/spec/mock-factory/index.ts (49 bytes)
CREATE sdk/sdk/testing/tsconfig.spec.json (474 bytes)
CREATE sdk/sdk/tsconfigs/tsconfig.jest.json (319 bytes)
CREATE sdk/sdk/tsconfigs/tsconfig.source.json (242 bytes)
CREATE sdk/sdk/tsconfigs/esm2020/tsconfig.jest.json (357 bytes)
CREATE sdk/sdk/tsconfigs/esm2020/tsconfig.json (127 bytes)
CREATE sdk/sdk/tsconfigs/esm2020/tsconfig.source.json (287 bytes)
CREATE sdk/sdk/.eslintignore (132 bytes)
CREATE sdk/sdk/.eslintrc.js (1075 bytes)
CREATE sdk/sdk/.gitignore (613 bytes)
CREATE sdk/sdk/package.json (4019 bytes)
⠏ Installing packages (npm)...
```

When the installation of packages is finished, I get again the same question:

```
✔ Packages installed successfully.
? 
Would you like to share anonymous data about the usage of Otter builders and schematics with the Otter Team at Amadeus ?
It will help us to improve our tools.
For more details and instructions on how to change these settings, see https://github.com/AmadeusITGroup/otter/blob/main/docs/telemetry/README.md.
         (Y/n) 
```

### Expected result

The question about enabling telemetry should be asked only once.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Ask only once to install telemetry when creating an sdk

## Related issues

- :bug: Fixes #1408

### CODE REVIEW

File: packages/@o3r/schematics/src/rule-factories/ng-add/dependencies.ts [hunk 1]
```diff
@@ -71,6 +71,8 @@ export interface SetupDependenciesOptions {
   runAfterTasks?: TaskId[];
   /** Callback to run after the task ID is calculated */
   scheduleTaskCallback?: (taskIds?: TaskId[]) => void;
+  /** Working directory */
+  workingDirectory?: string;
```
Comment (kpanot) [inline packages/@o3r/schematics/src/rule-factories/ng-add/dependencies.ts:75]: maybe you should indicate that this is for the installation process only (as the package.json edited is not affected by this parameter)

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@ama-sdk/create/src/index.ts [hunk 2]
```diff
@@ -69,26 +66,37 @@ const schematicArgs = [
...
+const resolveTargetDirectory = resolve(process.cwd(), targetDirectory);
 
 const run = () => {
+  let isSpecPathUrl = false;
+  try {
+    new url.URL(argv['spec-path']);
+    isSpecPathUrl = true;
+  } catch {
+    // If specPath is not an URL it will throw an error
+  }
```
Comment (kpanot) [inline packages/@ama-sdk/create/src/index.ts:81]:
```suggestion
  const isSpecPathUrl = url.URL.canParse(argv['spec-path']);
```

File: packages/@ama-sdk/create/src/index.ts [hunk 3]
```diff
@@ -69,26 +66,37 @@ const schematicArgs = [
...
+        ? [{ runner, args: ['set', 'version', getYarnVersion()], cwd: resolveTargetDirectory }]
         : []
     ),
-    ...schematicsToRun.slice(1).map(getSchematicStepInfo)
+    ...(argv['spec-path'] ? [{
+      args: [
+        binPath,
+        `${schematicsPackage}:typescript-core`,
+        ...schematicArgs,
+        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(join(process.cwd(), targetDirectory), argv['spec-path'])
```
Comment (kpanot) [inline packages/@ama-sdk/create/src/index.ts:96]:
```suggestion
        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(join(process.cwd(), targetDirectory), join(process.cwd(), argv['spec-path']))
```

**Review by vscaiceanu-1a — APPROVED** [review]

File: packages/@ama-sdk/create/src/index.ts [hunk 4]
```diff
@@ -69,26 +66,31 @@ const schematicArgs = [
...
+        ? [{ runner, args: ['set', 'version', getYarnVersion()], cwd: resolveTargetDirectory }]
         : []
     ),
-    ...schematicsToRun.slice(1).map(getSchematicStepInfo)
+    ...(argv['spec-path'] ? [{
+      args: [
+        binPath,
+        `${schematicsPackage}:typescript-core`,
+        ...schematicArgs,
+        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(join(process.cwd(), targetDirectory), join(process.cwd(), argv['spec-path']))
```
Comment (kpanot) [inline packages/@ama-sdk/create/src/index.ts:90]:
```suggestion
        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(resolveTargetDirectory, join(process.cwd(), argv['spec-path']))
```
(my bad, I could have seen it in my first suggestion)

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

**Pull request merged**